tests/queries/0_stateless/03272_partition_pruning_monotonic_func_bug.sql

```diff
@@ -0,0 +1,15 @@
+CREATE TABLE tt
+(
+    `id` Int64,
+    `ts` DateTime
+)
+ENGINE = MergeTree()
+ORDER BY dateTrunc('hour', ts)
+SETTINGS index_granularity = 8192;
+
+INSERT INTO tt VALUES (1, '2024-11-14 00:00:00'), (2, '2024-11-14 00:00:00');
+
+SELECT id
+FROM tt
+PREWHERE ts BETWEEN toDateTime(1731506400) AND toDateTime(1731594420);
+
```

Comment:
Let's assert via log, explain indexes or read/discarded granules that the key is used.